cmake/FindFuzzingLibrary.cmake

```diff
@@ -0,0 +1,45 @@
+# Copyright 2018 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# OSS_FUZZ provides its own fuzzing library in LIB_FUZZING_ENGINE environment
+# variable. For local fuzzing, search for the libFuzzer.a library that was
+# manually built.
+if(OSS_FUZZ)
+  set(FUZZING_LIBRARY_LOCATION $ENV{LIB_FUZZING_ENGINE})
```

Comment:
@wilhuff are we ok with reading from environment variables? OSS Fuzz builds the code inside a docker container with set environment variables.

Reply:
Reading from environment variables like this is fine, however I have some feedback:
  * The naming of this package confuses and interchanges the words "fuzzer" and "fuzzing".
  * It also uses "library" as a suffix when CMake convention is to drop "lib" and "library" from find-package names. For example, `libjpeg` is available in `FindJPEG.cmake`.

It's unfortunate that OSS_FUZZ calls this `LIB_FUZZING_ENGINE` but they're apparently abstracting over what this thing is. We should plan for this to be concrete and just happen to accept their variable to configure this.

So, assuming the thing we build against is "fuzzer":

  * `$ENV{LIB_FUZZING_ENGINE}` should just be the first hint to `find_library`.
    * If it's set, we'll use it, otherwise we'll fall back on whatever we've built.
    * I don't think we need to condition this on `OSS_FUZZ` being set.
  * Since we're finding "libfuzzer" this should in `FindFuzzer.cmake`
  * The output variable of `find_library` should be `FUZZER_LIBRARY`
  * The first argument to `find_package_handle_standard_args` should also match, so it should be `FUZZER`
  * That will result in `FUZZER_FOUND` being set.

Also, to compile against the fuzzer, don't we need headers? You should also `find_path` on one of the headers (like we do for [nanopb](https://github.com/firebase/firebase-ios-sdk/blob/master/cmake/FindNanopb.cmake#L17)) and attach that to the `INTERFACE_INCLUDE_DIRECTORIES` property of this target.

Reply:
I did the renaming and `find_library` modifications. As I explained in my previous comment, we do not need headers from libFuzzer as we only link to it and provide the fuzzing function that is needed by libFuzzer's main function